aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -366,18 +366,18 @@ Tensor& mvlgamma_(Tensor& self, int64_t p) {
 #define IMPLEMENT_UNARY_FLOATING_OP_VEC_CUDA(op)                       \
   IMPLEMENT_UNARY_FLOATING_OP_CORE(op)                                 \
   IMPLEMENT_UNARY_OP_OUT_INPLACE(op, cpu, CPU)                         \
-  IMPLEMENT_UNARY_OP_OUT_INPLACE(op, cpu, CUDA)
+  IMPLEMENT_UNARY_OP_OUT_INPLACE(op, cuda, CUDA)
 
 IMPLEMENT_UNARY_FLOATING_OP_VEC(atan)
 IMPLEMENT_UNARY_FLOATING_OP_VEC(cos)
 IMPLEMENT_UNARY_FLOATING_OP_VEC(cosh)
-IMPLEMENT_UNARY_OP_VEC(erf)
-IMPLEMENT_UNARY_OP_VEC(erfc)
-IMPLEMENT_UNARY_OP_VEC_CUDA(erfinv)
+IMPLEMENT_UNARY_FLOATING_OP_VEC(erf)
+IMPLEMENT_UNARY_FLOATING_OP_VEC(erfc)
+IMPLEMENT_UNARY_FLOATING_OP_VEC_CUDA(erfinv)
```

Comment:
All the ops (`atan, cos, cosh, erf, erfc, exp, tan, tanh`) using `IMPLEMENT_UNARY_FLOATING_OP_VEC` will throw error for: `torch.atan(torch.tensor(2, dtype=torch.int8, device="cuda"))` , `RuntimeError: _th_atan_out not supported on CUDAType for Char`.

Reply:
Also, since all the ops now use new macros (`IMPLEMENT_UNARY_FLOATING_OP_VEC`, `IMPLEMENT_UNARY_FLOATING_OP_VEC_CUDA`), should we remove these (https://github.com/pytorch/pytorch/pull/33322/commits/0d1e167bd0c95c85f0d1fa1966ac3334edafdf14#diff-a249164f42f7e57a9a0028bbc604ae77R351-R358) macros?